tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs

```diff
@@ -210,25 +220,26 @@ protected virtual void ExecuteOperations(IMongoClient client, Dictionary<string,
         {
             _objectMap = objectMap;
 
-            var factory = new JsonDrivenTestFactory(client, DatabaseName, CollectionName, bucketName: null, objectMap, eventCapturer);
+            var factory = CreateJsonFactory(client, DatabaseName, CollectionName, objectMap, eventCapturer);
 
             foreach (var operation in test[OperationsKey].AsBsonArray.Cast<BsonDocument>())
             {
                 ModifyOperationIfNeeded(operation);
                 var receiver = operation["object"].AsString;
                 var name = operation["name"].AsString;
-                var jsonDrivenTest = factory.CreateTest(receiver, name);
-
-                jsonDrivenTest.Arrange(operation);
-                if (test["async"].AsBoolean)
-                {
-                    jsonDrivenTest.ActAsync(CancellationToken.None).GetAwaiter().GetResult();
-                }
-                else
+                using (var jsonDrivenTest = factory.CreateTest(receiver, name))
```

Comment:
now, tests can dispose state after executing

Reply:
It would be better if this was not necessary and I think with some refactoring there won't be any need for tests to be disposable.